src/Apps/Order/Routes/Shipping/useAddressAutocomplete.ts

```diff
@@ -0,0 +1,145 @@
+import { AutocompleteInputOptionType } from "@artsy/palette"
+import { Address } from "Components/AddressForm"
+import { useFeatureFlag } from "System/useFeatureFlag"
+import { useDebounce } from "Utils/Hooks/useDebounce"
+import { getENV } from "Utils/getENV"
+import { useCallback, useEffect, useState } from "react"
+
+// NOTE: Due to the format of this key (a long string of numbers that cannot be parsed as json)
+// This key must be set in the env as a json string like SMARTY_EMBEDDED_KEY_JSON={ "key": "xxxxxxxxxxxxxxxxxx" }
+const { key: apiKey } = getENV("SMARTY_EMBEDDED_KEY_JSON") || { key: "" }
+
+export type ProviderSuggestion = {
+  city: string
+  entries: number
+  secondary: string
+  state: string
+  street_line: string
+  zipcode: string
+  source?: "postal" | "other"
+}
+
+export interface AddressSuggestion extends AutocompleteInputOptionType {
+  address: Omit<Address, "name">
+  entries: number
+}
+
+export const useAddressAutocomplete = (
+  address: Partial<Address> & { country: Address["country"] }
+) => {
+  const [result, setResult] = useState<ProviderSuggestion[]>([])
+  const isUSAddress = address.country === "US"
+  const isAPIKeyPresent = !!apiKey
+  const isFeatureFlagEnabled = !!useFeatureFlag("address_autocomplete_us")
+  const isAddressAutocompleteEnabled =
+    isAPIKeyPresent && isFeatureFlagEnabled && isUSAddress
+
+  useEffect(() => {
+    if (result.length > 0 && !isUSAddress) {
+      setResult([])
+    }
```

Comment:
Why wouldn't we just do this in the callback itself. This doesn't need to be an effect as far as I can see.

Reply:
The case here is when a user changes their country and then goes back - we don't want clicking into the address field to show them old suggestions. For now this is exclusively a US feature, so it's just US or not US, and if it isn't, we want to zero out the suggestions.